frontend/src/store/engagementSlice.ts

```diff
@@ -0,0 +1,56 @@
+import {
+  createEntityAdapter,
+  createSelector,
+  createSlice,
+  PayloadAction,
+} from '@reduxjs/toolkit';
+import toastr from 'toastr';
+
+import { Engagement } from '../model';
+import { get } from '../services/apiService';
+import { toDate } from '../utils/date';
+import { environment } from '../utils/environment';
+import { AppThunk, RootState } from './reducer';
+
+const { baseUrl } = environment;
+
+const engagementAdapter = createEntityAdapter<Engagement>();
+
+const engagementsFetchUrl = (start: Date, end: Date) =>
+  `${baseUrl}/staff/shifts?start=${start.toISOString()}&end=${end.toISOString()}`;
+
+export const engagementSlice = createSlice({
+  name: 'engagement',
+  initialState: engagementAdapter.getInitialState(),
+  reducers: {
+    setEngagements: (state, { payload }: PayloadAction<Engagement[]>) => {
+      engagementAdapter.setAll(state, payload);
+    },
+  },
+});
+
+export const { setEngagements } = engagementSlice.actions;
+
+export const fetchEngagements = (
+  start: Date,
+  end: Date,
+  token?: string
+): AppThunk => (dispatch) =>
+  get(engagementsFetchUrl(start, end), token)
+    .then((response) => response.json())
+    .then((response) => dispatch(setEngagements(response)))
+    .catch((err) => toastr.error(err));
```

Comment:
I know this is unchanged from before and not too important either, but this line can be shortened to
`.catch(toastr.error)`
I can't find a linting rule that would do that though. Either way, I'm fine with this staying the way it is.